### PR TITLE
[BEAM-4137] Divide elasticsearch pipeline options

### PR DESCRIPTION
**Pull request by lgajowy**
This PR moves Elasticsearch-related pipeline options to ElasticsearchIOITCommon class (the only place where those are needed). 

------------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

It will help us expedite review of your Pull Request if you tag someone (e.g. `@username`) to look at it.

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_GradleBuild/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_GradleBuild/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark_Gradle/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) </br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | --- | --- | --- | ---

### CODE REVIEW

**Comment by lgajowy** [comment]
Run Java PreCommit

**Comment by echauchot** [comment]
@lgajowy sure, I'll be glad to review the PR.

**Comment by lgajowy** [comment]
Thanks! 

It's probably better to wait with the review until I get the tests green... 😅

**Comment by lgajowy** [comment]
Run Java PreCommit

File: sdks/java/io/common/src/test/java/org/apache/beam/sdk/io/common/IOTestPipelineOptions.java [hunk 1]
```diff
@@ -21,31 +21,8 @@
...
-
-  void setElasticsearchServer(String value);
-
-  @Description("Http port for elasticsearch server")
-  @Default.Integer(9200)
-  Integer getElasticsearchHttpPort();
-
-  void setElasticsearchHttpPort(Integer value);
-
-  /* Solr */
```
[the remark this answers is not in the document]
Comment (echauchot) [reply sdks/java/io/common/src/test/java/org/apache/beam/sdk/io/common/IOTestPipelineOptions.java]: Besides, I only see a removal of Solr related options, but no addition

[the remark this answers is not in the document]
Comment (lgajowy) [reply sdks/java/io/common/src/test/java/org/apache/beam/sdk/io/common/IOTestPipelineOptions.java]: As for solr - I didn't find any usages of those options - there are no Solr tests nor any code that use it. It's a dead code so I deleted it. I isolated the removal in a separate commit. Isn't this enough?

**Comment by lgajowy** [comment]
@echauchot phrase triggering is impossible now on the current codebase. It would be possible only when a Jenkins job exists that runs the elasticsearch tests. There is no such job yet - this is another task [that JB is on](https://issues.apache.org/jira/browse/BEAM-4398). I just refactored here simply to finish BEAM-4137. However, FWIW, I tested this locally and it reads the options well. Please let me know if such test is enough.

File: sdks/java/io/common/src/test/java/org/apache/beam/sdk/io/common/IOTestPipelineOptions.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (echauchot) [reply sdks/java/io/common/src/test/java/org/apache/beam/sdk/io/common/IOTestPipelineOptions.java]: I did not know it was dead code. You were right to remove it. Yes a separate commit is enough of course.

**Comment by echauchot** [comment]
@lgajowy if you have run the integration tests locally, then it is fine for me. Indeed, we will need to create missing jenkins jobs for ITs.

**Pull request merged**